src/Octoshift/BaseCommandArgs.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Linq;
+using System.Text;
+
+namespace OctoshiftCLI
+{
+    public abstract class BaseCommandArgs
+    {
+        [LogName("VERBOSE")]
+        public bool Verbose { get; set; }
+
+        public abstract void Validate();
+
+        public void Log(OctoLogger log)
+        {
+            if (log is null)
+            {
+                throw new ArgumentNullException(nameof(log));
+            }
+
+            log.Verbose = Verbose;
+            var sb = new StringBuilder();
+
+            foreach (var property in GetType().GetProperties())
```

Comment:
Might be better to get all `public` properties

Reply:
the default behavior of GetProperties is to only return public properties